packages/client/src/internals/httpRequest.ts

```diff
@@ -59,17 +59,25 @@ export function httpRequest<TResponseShape = TRPCResponse>(
     const url = getUrl();
 
     Promise.resolve(rt.headers())
-      .then((headers) =>
-        rt.fetch(url, {
+      .then((rawHeaders) => {
+        const headers: HeadersInit = { 'content-type': 'application/json' };
+        for (const key in rawHeaders) {
+          if (rawHeaders[key] !== undefined) {
+            if (Array.isArray(rawHeaders[key])) {
+              headers[key] = (rawHeaders[key] as string[]).join(',');
+            } else {
+              headers[key] = rawHeaders[key] as string;
```

Comment:
You can remove TS assertion by using a local reference:
```suggestion
          const value = rawHeaders[key];
          if (value !== undefined) {
            if (Array.isArray(value)) {
              headers[key] = value.join(',');
            } else {
              headers[key] = value;
```

Reply:
You can remove the last one by setting header before the undefined check